internal/app/handlers.go

```diff
@@ -795,6 +796,20 @@ func (app *App) syncGetRootV3(c *gin.Context) {
 	})
 }
 
+func (app *App) syncGetRootV3(c *gin.Context) {
+	app.syncGetRoot(c, func(c *gin.Context) {
+		c.JSON(http.StatusNotFound, gin.H{"message": "root not found"})
+	})
+}
+
+func (app *App) syncGetRootV4(c *gin.Context) {
+	app.syncGetRoot(c, func(c *gin.Context) {
+		c.JSON(http.StatusOK, messages.SyncRootV3Response{
```

Comment:
```suggestion
		c.JSON(http.StatusOK, messages.SyncRootV3Request{
```

Got it to work after changing this line.

Reply:
It doesn't make sense to respond with a Request. It must be something else.

The difference will be:

```json
{
    "generation": 0,
    "schemaVersion": 0
}
```

to:

```json
{
    "generation": 0,
    "broadcast": false
}
```

Perhaps adding the correct `schemaVersion` as seen in the rest of the code?